### ISSUE TITLE
Add type conversion to convert data types to supported data types in ttnn

### ISSUE THREAD

**Issue by mtopalovicTT** [tenstorrent/tt-mlir#1277]
ttnn doesn't support all data types which come from forge. One example is is int32 is mapped to Uint32
This can lead to some strange IR like this

`to_layout(%arg1) { dtype = Uint32 }(tensor<10x10xi32) -> tensor<10x10xi32)`

what happens is that during lowering to ttnn when creating dtype attribute for `to_layout` we convert outputTensor data type to metal supported data type.

Would it be possible to create type conversion which will do all the conversions upfront to ttnn supported data types?

Would it have implications on forge?

**Comment by mtopalovicTT** [comment]
cc @svuckovicTT @nsmithtt @pilkicTT

**Comment by nsmithtt** [comment]
@mtopalovicTT, I think TTIR should be flexible and accept non-hw types like ints and bools and complex types etc.  We should handle this kind of data format conversion during a TTIR->TTIR pass.

@mrakitaTT, @AleksKnezevic, any input from the stablehlo side?  There have been many similar df conversion situations there.

**Comment by AleksKnezevic** [comment]
Yeah, there were a few cases where we've had to convert unsupported types. @mmanzoorTT did the conversion, so he can comment more.

**Comment by mtopalovicTT** [comment]
> @mtopalovicTT, I think TTIR should be flexible and accept non-hw types like ints and bools and complex types etc. We should handle this kind of data format conversion during a TTIR->TTIR pass.

@nsmithtt sure, that was my first proposal. I pinged pedrag ysterday just to double check what does he think about these approaches.

**Comment by sdjordjevicTT** [comment]
Thanks, @svuckovicTT for the notes, I think we captured all the main points from our discussion.

**Comment by skhatriTT** [comment]
@mtopalovicTT what is our status on this? correct me if I am wrong but it looks like we have a temporary fix in place
